src/orquestra/quantum/wip/operators/_pauli_operators.py

```diff
@@ -0,0 +1,546 @@
+    if not isinstance(object, (PauliSum, PauliTerm, int, float, complex)):
+        raise TypeError(
+            f"Can't carry out operation with object of type {type(object)}."
+        )
+
+
+class PauliTerm:
+    """
+    A datastructure for storing information about a single Pauli Term
+    """
+
+    def __init__(
+        self,
+        operator: Union[Tuple[str, int], str],
+        coefficient: CoefficientTypes = 1.0,
+    ):
+        if isinstance(operator, str):
+            matched_pattern = re.match(r"([a-zA-Z]+)([0-9]+)", operator, re.I)
+
+            if not matched_pattern:
+                raise ValueError("Badly formatted string representation passed.")
+
+            op, qubit_idx = matched_pattern.groups()
+        elif isinstance(operator, tuple):
+            op, qubit_idx = operator
+
+        qubit_idx = int(qubit_idx)
+
+        if op not in allowed_operators:
+            raise ValueError(f"Passed operator not supported. Got {op}.")
+
+        self._ops: Dict[int, str] = OrderedDict()
+        if op != "I":
+            self._ops[qubit_idx] = op
+
+        self.coefficient = complex(coefficient)
+
+    @classmethod
+    def from_list(
+        cls,
+        list_of_terms: List[Tuple[str, int]],
+        coefficient: CoefficientTypes = 1.0,
+    ) -> "PauliTerm":
+        """
+        A slightly more efficient constructor when all the elements of the term are
+         known beforehand. Users should employ this function instead of creating
+         individual terms and multiplying.
+        """
+
+        ############### Some checks on input first ###############
+        op_list, idx_list = zip(*list_of_terms)
+
+        if not all([isinstance(op, tuple) for op in list_of_terms]):
+            raise ValueError(
+                "The list can only contain tuples of the form (op, index). If you"
+                " want to initialize from strings, check the"
+                " PauliTerm.from_str function."
+            )
+
+        valid_input = all(
+            [
+                all([op in allowed_operators for op in op_list]),
+                all([idx >= 0 for idx in idx_list]),
+            ]
+        )
+
+        if not valid_input:
+            raise ValueError(
+                "Some of the input was invalid. Make sure you only pass Pauli operators"
+                " and non-negative qubit indices."
+            )
+
+        if len(set(idx_list)) != len(idx_list):
+            raise ValueError(
+                "Duplicate indices used in list. Manually create terms"
+                " and multiply them instead."
+            )
+
+        ##########################################################
+
+        result_term = PauliTerm.identity()
+        result_term.coefficient = complex(coefficient)
+        for op, idx in list_of_terms:
+            if op != "I":
+                result_term._ops[idx] = op
+
+        return result_term
+
+    @classmethod
+    def from_str(cls, str_pauli_term: str) -> "PauliTerm":
```

Comment:
Both `from_str` and `from_list` methods are a bit suspicious to me, mainly because they access the internal dictionary of PauliTerm. In most cases, this means, that the `__init__` method needs some attention. WDYT about making the `__init__` method accept the dictionary containing operators and qubit indices, and the coefficient, while te static factories would just parse the incoming data and pass them over to the initializer?